hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java

```diff
@@ -144,9 +145,9 @@ public void handleQuery(@NonNull final Bytes requestBuffer, @NonNull final Buffe
         final var function = functionOf(query);
 
         final var handler = dispatcher.getHandler(query);
-        final var queryHeader = handler.extractHeader(query);
+        var queryHeader = handler.extractHeader(query);
         if (queryHeader == null) {
-            throw new StatusRuntimeException(Status.INVALID_ARGUMENT);
+            queryHeader = QueryHeader.DEFAULT;
```

Comment:
I might suggest:

```
    final var queryHeader = requireNonNullElse(handler.extractHeader(query), QueryHeader.DEFAULT);
```

Personally, I like using final where possible. It just feels "tighter" -- fewer loose ends.

Reply:
Thanks for the suggestion, that looks cleaner. I see that you already approved. If I end up making changes for other reasons then I will incorporate this suggestion.